app/src/main/java/com/babylon/wallet/android/presentation/transaction/composables/TransactionPreviewHeader.kt

```diff
@@ -50,57 +54,60 @@ fun TransactionPreviewHeader(
     scrollState: ScrollState
 ) {
     val context = LocalContext.current
-
-    // TODO improve this at later time.
-    // AnimationRangePx is the threshold which scrollState.value hits and then transition of the motion layout starts.
-    // When its relatively low i.e. 40.dp we start transition early and if content is large enough scroll state continue
-    // with its scroll value.
-    // When content is relatively small we might start topbar transition too early which makes scroll state to reset its
-    // scroll.value which is causing the flickering.
-    // That is why for smaller content, we increase animation threshold value from 40.do to 200.dp
-    val animationValue = remember(scrollState.maxValue) {
-        if (scrollState.maxValue >= 200) {
-            40.dp
-        } else {
-            200.dp
-        }
+    val motionSceneContent = remember {
+        context.resources
+            .openRawResource(R.raw.transaction_review_top_bar_scene)
+            .readBytes()
+            .decodeToString()
     }
 
-    val animationRangePx = with(LocalDensity.current) {
-        animationValue.toPx()
-    }
-    val progress by remember(scrollState.value) {
-        if (scrollState.maxValue != Int.MAX_VALUE) {
-            derivedStateOf {
+    // size of the preview header
+    var size by remember { mutableStateOf(IntSize.Zero) }
+
+    val animationRangePx = with(LocalDensity.current) { 200.dp.toPx() }
+    val progress by remember {
+        derivedStateOf {
+            // if max value of the scroll is less than the height of the preview header then don't use the animation
+            // the + 100 is for these cases where the warnings at the bottom of the receipt are visible (e.g. congested network)
+            if (scrollState.maxValue <= size.height + 100) {
```

Comment:
If I don't add the +100 it flickers again. And I am wondering why this happens. 🤷🏽‍♂️ 
<img width="200" alt="Screenshot 2024-01-31 at 1 10 53 PM" src="https://github.com/radixdlt/babylon-wallet-android/assets/118305718/9be09597-424e-4f34-b29f-363427b1b03f">